sklearn/tests/test_common.py

```diff
@@ -644,3 +644,24 @@ def test_global_output_transform_pandas(estimator):
     _set_checking_parameters(estimator)
     with ignore_warnings(category=(FutureWarning)):
         check_global_ouptut_transform_pandas(estimator.__class__.__name__, estimator)
+
+
+STATELESS_ESTIMATORS = [
+    est for est in _tested_estimators() if _safe_tags(est, key="stateless")
+]
+
+
+@pytest.mark.parametrize(
+    "estimator", STATELESS_ESTIMATORS, ids=_get_check_estimator_ids
+)
+def test_no_fitted_error_stateless_estimator(estimator):
+    """
+    Check that using transform or predict without prior fitting
+    doesn't raise a NotFittedError.
+    """
+    X, _ = make_blobs(n_samples=80, n_features=4, random_state=0)
+    X = X[X > 0].reshape(-1, 1)
+
+    if hasattr(estimator, "transform"):
+        # Does not raise
+        estimator.transform(X)
```

Comment:
It is true that we expect a transformer now. So either we add it to the `_yield` for transformer or we think that estimator like `DummyClassifier` or `DummyRegressor` could become somehow stateless (with some parameter combination) in which case we want to try both `transform` and `predict`.

Reply:
I think that we want to check the length of the transform to check that we still have `n_samples`.

Reply:
Thanks for these detailed explanations. Regarding your option number 2, wouldn't adding `predict` raise code coverage concerns since this line won't be tested?

Reply:
I am fine with keeping it for transformers for the moment. If we tackle "predictors" then we will make the change at this moment.